docs/algorithms.md

```diff
@@ -84,7 +90,33 @@ To learn more about Guided GradCAM, visit the following resources:
 - [Original paper](https://arxiv.org/abs/1610.02391)
 - [Website](http://gradcam.cloudcv.org/)
 
-Note: this method is currently available in master and will be available in pip / conda releases soon.
+### Feature Ablation
+Feature ablation is a perturbation based approach to computing attribution, involving replacing each input feature with a given baseline / reference value (e.g. 0), and computing the difference in output. Input features can also be grouped and ablated together rather than individually.
+This can be used in cases such as images, where an entire segment or region can be ablated, measuring the importance of the segment (feature group).
+        
+
+### Feature Permutation
+Feature permutation is a perturbation based approach which takes each feature individually, randomly permutes the feature values within a batch and computes the change in output (or loss) as a result of this modification. Like feature ablation, input features can also be grouped and shuffled together rather than individually.
+
+To learn more about Feature Permutation, visit the following resources:  
+- [Interpretable ML Book](https://christophm.github.io/interpretable-ml-book/feature-importance.html)
+
+### Occlusion
+Occlusion is a perturbation based approach to compute attribution, involving replacing each contiguous rectangular region with a given baseline / reference, and computing the difference in output. For features located in multiple regions (hyperrectangles), the corresponding output differences are averaged to compute the attribution for that feature. Occlusion is most useful in cases such as images, where pixels in a contiguous rectangular region are likely to be highly correlated.
```

Comment:
nit: sometimes we say to computing sometimes to compute. it would be good to be consistent and call it to compute.